server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -383,21 +393,68 @@ private void snapshot(
             GatedCloseable<IndexCommit> wrappedSnapshot = null;
             try {
-                // we flush first to make sure we get the latest writes snapshotted
-                wrappedSnapshot = indexShard.acquireLastIndexCommit(true);
-                final IndexCommit snapshotIndexCommit = wrappedSnapshot.get();
-                repository.snapshotShard(
-                    indexShard.store(),
-                    indexShard.mapperService(),
-                    snapshot.getSnapshotId(),
-                    indexId,
-                    wrappedSnapshot.get(),
-                    getShardStateId(indexShard, snapshotIndexCommit),
-                    snapshotStatus,
-                    version,
-                    userMetadata,
-                    ActionListener.runBefore(listener, wrappedSnapshot::close)
-                );
+                if (remoteStoreIndexShallowCopy && indexShard.indexSettings().isRemoteStoreEnabled()) {
+                    long startTime = threadPool.absoluteTimeInMillis();
+                    // we flush first to make sure we get the latest writes snapshotted
+                    wrappedSnapshot = indexShard.acquireLastIndexCommitAndRefresh(true);
+                    long primaryTerm = indexShard.getOperationPrimaryTerm();
+                    final IndexCommit snapshotIndexCommit = wrappedSnapshot.get();
+                    long commitGeneration = snapshotIndexCommit.getGeneration();
+                    indexShard.acquireLockOnCommitData(snapshot.getSnapshotId().getUUID(), primaryTerm, commitGeneration);
+                    try {
+                        repository.snapshotRemoteStoreIndexShard(
+                            indexShard.store(),
+                            snapshot.getSnapshotId(),
+                            indexId,
+                            wrappedSnapshot.get(),
+                            getShardStateId(indexShard, snapshotIndexCommit),
+                            snapshotStatus,
+                            primaryTerm,
+                            startTime,
+                            ActionListener.runBefore(listener, wrappedSnapshot::close)
+                        );
+                    } catch (IndexShardSnapshotFailedException e) {
+                        logger.error(
+                            "Shallow Copy Snapshot Failed for Shard ["
+                                + indexId.getName()
+                                + "]["
+                                + shardId.getId()
+                                + "] for snapshot "
+                                + snapshot.getSnapshotId()
+                                + ", releasing acquired lock from remote store"
+                        );
+                        indexShard.releaseLockOnCommitData(snapshot.getSnapshotId().getUUID(), primaryTerm, commitGeneration);
+                        throw e;
+                    }
+                    long endTime = threadPool.absoluteTimeInMillis();
```

Comment:
Use `relativeTimeInMillis `. Note this method uses a cached time which will cause a discrepancy of the order of 200ms